c/meterpreter/source/extensions/priv/ms-rprn_h.h

```diff
@@ -0,0 +1,317 @@
+
+
+/* this ALWAYS GENERATED file contains the definitions for the interfaces */
+
+
+ /* File created by MIDL compiler version 8.01.0622 */
+/* at Tue Jan 19 14:14:07 2038
+ */
+/* Compiler settings for ms-rprn.idl:
+    Oicf, W1, Zp8, env=Win64 (32b run), target_arch=AMD64 8.01.0622 
+    protocol : all , ms_ext, c_ext, robust
+    error checks: allocation ref bounds_check enum stub_data 
+    VC __declspec() decoration level: 
+         __declspec(uuid()), __declspec(selectany), __declspec(novtable)
+         DECLSPEC_UUID(), MIDL_INTERFACE()
+*/
+/* @@MIDL_FILE_HEADING(  ) */
```

Comment:
Based on this comment, is this possible to extract into a preprocessing step that generates this?  We would need to keep in mind needs for `mingw` builds.

Or possibly reduce this to the required methods?

Another though from @zeroSteiner may be to reuse these file from mimikatz.

Reply:
Should be resolved in 174ae1a where I updated it to reuse the components provided by mimikatz. Waiting on automation to validate the mingw build now.

Reply:
Nevermind, forgot that automation didn't test the mingw builds. So I tested that with and without my changes, in both cases it was broken. Commit 229637b fixed the build process so both x86 and x64 binaries are built for the priv extension.

I just need to test this a bit more thoroughly and then I should be able to get it landed. I've already tested it on Windows 10 and it worked as intended but I'd like to try some older version of Windows too.